agent/php_txn.c

```diff
@@ -1067,7 +1070,16 @@ nr_status_t nr_php_txn_end(int ignoretxn, int in_post_deactivate TSRMLS_DC) {
    * by calling newrelic_end_transaction inside nested function scopes)
    * the stack of stacked segments has to be cleaned up.
    */
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+    && !defined OVERWRITE_ZEND_EXECUTE_DATA
+  for (nr_segment_t* segment = nr_txn_get_current_segment(NRPRG(txn), NULL);
+       segment != NULL && segment != NRTXN(segment_root);
+       segment = nr_txn_get_current_segment(NRPRG(txn), NULL)) {
+    nr_segment_end(&segment);
+  }
```

Comment:
A `while` loop seems like a more appropriate loop here
```suggestion
  nr_segment_t* segment = nr_txn_get_current_segment(NRPRG(txn), NULL);
  while(segment != NULL && segment != NRTXN(segment_root) {
    nr_segment_end(&segment);
    segment = nr_txn_get_current_segment(NRPRG(txn), NULL);
  }
```

Reply:
I don't see a compelling reason for either loop over the other. Is there a big reason to use `while `here?

Reply:
In my opinion a `while` loop in this situation makes the loop conditions easier to read and is typically more appropriate when you aren't sure how many elements you're looping over.